### ISSUE TITLE
Show .epub files in YACReaderLibrary?

### ISSUE THREAD

**Issue by rafaelgonzalez** [YACReader/yacreader#137]
Hi there.

Publishers often provide DRM-free comics in multiple formats  like `.cbz`, `.pdf`, `.epub`, etc.

YACReader is perfectly capable of opening `.epub` files, yet YACReaderLibrary does not show `.epub` files at all on Mac OSX. I can't speak for the versions for other operating systems.

Any reason for this? 

Some of my comics have the highest image quality in the `.epub` files only. I would hate to have to keep lower quality files.

I have searched both here and the forum and there is barely any mention of epub files.

Cheers!

**Comment by selmf** [comment]
> YACReader is perfectly capable of opening `.epub` files, yet YACReaderLibrary does not show `.epub` files at all on Mac OSX. I can't speak for the versions for other operating systems.

Except it isn't. We have no epub support. Some files may work because the publisher actually does not take full advantage of the file format, which makes it just a zip file with images, but any real epub won't work.
 
> Any reason for this?

As written above, we have no support for epub so the library does not scan for these files.

**Comment by rafaelgonzalez** [comment]
Fair enough, that explains why you would be adverse to providing proper EPUB support, as that would go beyond the scope of just comics.

A shame, I'll have to convert those `.epub` to another format then. Thanks.

**Comment by selmf** [comment]
Sorry if I came around a little harsh. We actually are interested in having .epub support. The issue is that there is no simple drop-in library with a suitable Open Source license we can use, so it isn't trivial to add it. For pdf, we have a rendering library and the comic book format already provides us with images, so YACReader basically is a better image viewer.
For epub, we would need to write our own renderer. This is doable as epub is built upon web technology and we can use other libraries as the building blocks, but it requires a fair amount of work. I'm sure we already have some discussion regarding that topic on our forum, but it probably wouldn't hurt to have a feature request here too, so feel free to open one ;)

**Comment by rafaelgonzalez** [comment]
> Sorry if I came around a little harsh.

Oh no worries, you didn't come through harsh at all! 🤗

The joys of conveying emotion and tone online. 😉 

> We actually are interested in having .epub support. The issue is that there is no simple drop-in library with a suitable Open Source license we can use, so it isn't trivial to add it. For pdf, we have a rendering library and the comic book format already provides us with images, so YACReader basically is a better image viewer.
>
> For epub, we would need to write our own renderer. This is doable as epub is built upon web technology and we can use other libraries as the building blocks, but it requires a fair amount of work.

Thanks for the further explanation. This makes a ton of sense.

> I'm sure we already have some discussion regarding that topic on our forum, but it probably wouldn't hurt to have a feature request here too, so feel free to open one ;)

I would like to re-iterate that regarding the forums, I could find almost no mention of EPUB at all, based on a simple search: <https://yacreader.com/forum/search?query=epub&childforums=1>. Not a critique, just so you're aware.

In any case, I'm definitely happy if you want to leave this issue opened; I closed it assuming there was no intent to support EPUB in the future. Great to know otherwise!

**Comment by rafaelgonzalez** [comment]
Submitted #140, since it's a prerequisite for this issue (#137)

**Comment by selmf** [comment]
What we actually could do is add a setting to optionally just treat .epub as a .cbz, with a warning that this will only work for some books and will horribly fail on others. @luisangelsm has mentioned his interest in adding a similar feature for PDF files as comic publishers often use that format as an image container without using its actual features too.

**Comment by rafaelgonzalez** [comment]
Speaking for myself, I'd be totally happy with that sort of solution.

**Comment by luisangelsm** [comment]
Image based comics in epub format are now supported. Full epub compatibility is a bit out of scope at the moment.
